snuba/cli/multistorage_consumer.py

```diff
@@ -79,6 +80,9 @@
     "--output-block-size", type=int,
 )
 @click.option("--log-level")
+@click.option(
+    "--dead-letter-topic", help="Dead letter topic to send failed insert messages."
+)
```

Comment:
We should add these options to the regular consumer as well. Not only the multi storage consumer. That makes switching from single consumer to multi storage consumer much less likely to fail.

For example, the consumer which fills the profiles dataset would be a single storage consumer but could use this functionality when it first starts off in production.

Reply:
Why though? The single storage consumer never drops data. If the write fails it does not commit and crashes, so it will retry. This dead letter queue implementation is only useful for multi storage consumer for the secondary storages if we do not crash the consumer upon error

Reply:
I was thinking about the manner in which we introduce new datasets and their consumers in production. When they are introduced, we cannot foresee if and when they could fail. In those cases, should a single instance of write failure cause the consumer to crash and cause backlog of the system? Or should it keep moving along and let dead letter consumer deal with retrying the failures.

The write failures we were seeing while writing to the new errors_v2 cluster were a storage problem. Not a consumer related problem. 

We can go with only multi-storage consumers for now since it increases complexity of the system for little gain for now.